test/strategies/elkilpstrategy.js

```diff
@@ -0,0 +1,308 @@
+
+            // We deposit into the staking contract our ELP tokens
+            await elkIlpStrategyV5.deposit(BigNumber.from((await elkPairContract.balanceOf(owner.address)).toString()))
+            // We get the 'totalDeposit' after our deposit into the strategy
+            const totalDepositsAfter = (await elkIlpStrategyV5.totalDeposits()).toString();
+
+            // We check all balances
+            expect((await elkIlpStrategyV5.balanceOf(owner.address)).toString()).to.equal(amount)
+            expect((await stakingContract.balanceOf(elkIlpStrategyV5.address)).toString()).to.equal(amount)
+            expect(BigNumber.from(totalDepositsAfter)).gt(BigNumber.from(totalDepositsBefore));
+            expect((await stakingContract.coverageOf(elkIlpStrategyV5.address)).toString()).to.equal("0")
+
+            // We impersonate the 'owner' of the WAVAX-ELK StakingRewardsILP contract
+            await ethers.provider.send('hardhat_impersonateAccount', ['0xba49776326A1ca54EB4F406C94Ae4e1ebE458E19']);
+            const admin = await ethers.provider.getSigner('0xba49776326A1ca54EB4F406C94Ae4e1ebE458E19')
+
+            const stakingcontract = await ethers.getContractAt(staking_contract, elpStakingRewardAddress, admin);
+            // We set the coverage amount for our Strategy
+            await stakingcontract.setCoverageAmount(elkIlpStrategyV5.address, 1000000000000);
+
+            await hre.network.provider.request({
+                method: "hardhat_stopImpersonatingAccount",
+                params: ["0xba49776326A1ca54EB4F406C94Ae4e1ebE458E19"],
+            });
+
+            // We check the amount of coverage the Strategy contract address currently has
+            expect((await stakingContract.coverageOf(elkIlpStrategyV5.address)).toString()).to.equal('1000000000000')
+            // expect((await elk_ilp_strategy_v5.checkCoverage()).toString()).to.equal('1000000000000')
+
+            // We should check now if, upon withdrawal, we get a bigger amount of ELP tokens
+            // We withdraw from the strategy all tokens we deposited
+            await elkIlpStrategyV5.withdraw(BigNumber.from((await elkIlpStrategyV5.balanceOf(owner.address)).toString()))
+            const totalDepositsAfterWithdraw = (await elkIlpStrategyV5.totalDeposits()).toString();
+
+            // We check if balances are correct
+            // Since we have only one depositor, strategy contract and staking contract should both be back at 0
+            expect((await elkIlpStrategyV5.balanceOf(owner.address))).eq(0)
+            expect((await stakingContract.balanceOf(elkIlpStrategyV5.address))).eq(0)
+            // The total deposit should be back to its previous value
+            expect(BigNumber.from(totalDepositsAfterWithdraw)).eq(BigNumber.from(totalDepositsBefore));
+            // We should get a slightly bigger amount of ELP tokens since we got the coverage as well
+            expect(await elkPairContract.balanceOf(owner.address)).gt(amount);
```

Comment:
ok so here we're verifying that you actually got more than you previsously had without any reinvest, which is good. I'd advise you to add an extra check to verify that the strategyv5 does not have any elk dust, so the balance of elk of the elkIlpStrategyV5 should be 0. Just be mindful that reinvest can and will leave dust sometimes, but you're not calling reinvest here

Reply:
It does leave some elk dust indeed . The reinvest is called in the `withdraw` function

Reply:
we would need a way to understand if that dust is from the elk il protection or if it's just the dust from the reinvest. Also would be good to understand how big is the dust, is it a considerable amount?

Reply:
Did multiple runs and I got in between 49115060544 and 72922815817 ELK (18 Decimals)